### ISSUE TITLE
label   for negative!

### ISSUE THREAD

**Issue by feixiangdekaka** [bubbliiiing/retinanet-keras#1]
# 计算每一个先验框应该有的权重
        alpha_factor_for_back = keras.backend.ones_like(labels_for_back) * alpha
        alpha_factor_for_back = backend.where(keras.backend.equal(labels_for_back, 1), alpha_factor_for_back, 1 - alpha_factor_for_back)
        focal_weight_for_back = backend.where(keras.backend.equal(labels_for_back, 1), 1 - classification_for_back, classification_for_back)
        focal_weight_for_back = alpha_factor_for_back * focal_weight_for_back ** gamma

这个labels_for_back 中哪个元素是1，  负样本如何在制作标签中指定1，还是都设为0

**Comment by bubbliiiing** [comment]
负样本不是没有存在目标的先验框么，那他的labels_for_back就是0吧

**Comment by bubbliiiing** [comment]
就全都是0

**Comment by feixiangdekaka** [comment]
> 负样本不是没有存在目标的先验框么，那他的labels_for_back就是0吧

keras.backend.equal(labels_for_back, 1)  那这个输出全是False，然后取
alpha_factor_for_back = 1 - alpha_factor_for_back 
focal_weight_for_back=classification_for_back作为系数 可以这样理解吗

**Comment by bubbliiiing** [comment]
是的
